src/providers/create/ProvidersCreatePage.tsx

```diff
@@ -0,0 +1,178 @@
+import { validateProviderAndSecret } from '../utils/validators/validateProviderAndSecret';
+
+import ProviderCreateActionsSection from './components/ProviderCreateActionsSection';
+import ProvidersCreateFormsSection from './components/ProvidersCreateFormsSection';
+import ProvidersCreatePageHeader from './components/ProvidersCreatePageHeader';
+import { createProvider } from './utils/createProvider';
+import { createProviderSecret } from './utils/createProviderSecret';
+import { patchProviderSecretOwner } from './utils/patchProviderSecretOwner';
+import { providerTemplate } from './utils/providerTemplate';
+import { secretTemplate } from './utils/secretTemplate';
+
+import './ProvidersCreatePage.style.scss';
+
+const ProvidersCreatePage: FC<{
+  namespace: string;
+}> = ({ namespace }) => {
+  const { t } = useForkliftTranslation();
+  const history = useHistory();
+  const [isLoading, toggleIsLoading] = useToggle();
+  const [activeNamespace, setActiveNamespace] = useActiveNamespace();
+  const [providerNames, providerNamesLoaded] = useK8sWatchProviderNames({ namespace });
+
+  const defaultNamespace = getDefaultNamespace();
+  const initialProjectName =
+    activeNamespace === Namespace.AllProjects ? defaultNamespace : activeNamespace;
+  const initialNamespace = namespace || initialProjectName || defaultNamespace;
+  const [apiError, setApiError] = useState<Error | null>();
+  const [validationError, setValidationError] = useState<ValidationMsg>({
+    msg: 'Missing provider name',
+    type: ValidationState.Error,
+  });
+  const [newProvider, setNewProvider] = useState<V1beta1Provider>({
+    ...providerTemplate,
+    metadata: {
+      ...providerTemplate.metadata,
+      namespace: initialNamespace,
+    },
+  });
+  const [newSecret, setNewSecret] = useState<IoK8sApiCoreV1Secret>({
+    ...secretTemplate,
+    metadata: {
+      ...secretTemplate.metadata,
+      namespace: initialNamespace,
+    },
+  });
+  const [projectName, setProjectName] = useState<string>(initialProjectName);
+
+  if (!newSecret) {
+    return <span className="text-muted">{t('No credentials found.')}</span>;
+  }
+
+  const onNewSecretChange = (newValue: IoK8sApiCoreV1Secret) => {
+    setApiError(null);
+    setValidationError(validateProviderAndSecret(newProvider, newValue, providerNames));
+    setNewSecret(newValue);
+  };
+
+  const onNewProviderChange = (newValue: V1beta1Provider) => {
+    setApiError(null);
+    setValidationError(validateProviderAndSecret(newValue, newSecret, providerNames));
+    // Sync secret with new URL
+    const updatedSecret = {
+      ...newSecret,
+      data: { ...newSecret.data, url: encode(newValue?.spec?.url ?? '') },
+    };
+    setNewProvider(newValue);
+    setNewSecret(updatedSecret);
+  };
+
+  const onProjectNameChange = (newValue: string) => {
+    setApiError(null);
+    setValidationError(validateProjectName(newValue));
+    setProjectName(newValue);
+  };
+
+  const onUpdate = async () => {
```

Comment:
why the split? each one has try and catch and u return in a catch.. use a single try catch